### PR TITLE
DM-370 add missing resource type/validation

### PR DESCRIPTION
**Pull request by sutherlanda**

### CODE REVIEW

File: src/shared/lib/validation/opportunity/sprint-with-us.ts [hunk 1]
```diff
@@ -46,14 +46,39 @@ export function validateSWUOpportunityPhaseMaxBudget(raw: number, maxTotalBudget
...
+}
+
+export function validateFullTime(raw: any): Validation<boolean> {
+  return isBoolean(raw) ? valid(raw) : invalid(['You must provide a boolean value.']);
+}
+
+export function validatePhaseRequiredCapabilities(raw: any): ArrayValidation<CreateSWUOpportunityPhaseRequiredCapabilityBody, CreateSWUOpportunityPhaseRequiredCapabilityErrors> {
+  if (!isArray(raw)) { return invalid([{ parseFailure: ['Please provide an array of required capabilities.'] }]); }
+  return validateArrayCustom(raw, validatePhaseRequiredCapability, {});
+}
```
Comment (dhruvio) [inline src/shared/lib/validation/opportunity/sprint-with-us.ts]: Validate that at least one capability is provided.

File: src/shared/lib/resources/opportunity/sprint-with-us.ts [hunk 2]
```diff
@@ -123,9 +132,12 @@ export type CreateSWUOpportunityStatus
   | SWUOpportunityStatus.UnderReview
   | SWUOpportunityStatus.Draft;
 
-export interface CreateSWUOpportunityPhaseBody extends Omit<SWUOpportunityPhase, 'createdAt' | 'createdBy' | 'updatedAt' | 'updatedBy' | 'startDate' | 'completionDate'> {
+export type CreateSWUOpportunityPhaseRequiredCapabilityBody = Omit<SWUOpportunityPhaseRequiredCapability, 'createdAt' | 'createdBy' | 'updatedAt' | 'updatedBy'>;
```
Comment (dhruvio) [inline src/shared/lib/resources/opportunity/sprint-with-us.ts]: More a nitpick than anything — using a `Pick` here would be much easier to read: `Pick<SWUOpportunityPhaseRequiredCapability, 'capability' | 'fullTime'>`

Comment (dhruvio) [reply src/shared/lib/resources/opportunity/sprint-with-us.ts]: Optional change

File: src/shared/lib/validation/opportunity/sprint-with-us.ts (hunk 1 above)
Comment (sutherlanda) [reply src/shared/lib/validation/opportunity/sprint-with-us.ts]: Done, thanks.

File: src/shared/lib/resources/opportunity/sprint-with-us.ts (hunk 2 above)
Comment (sutherlanda) [reply src/shared/lib/resources/opportunity/sprint-with-us.ts]: Good suggestion, thanks.  Done.

**Pull request merged**